public/app/plugins/datasource/prometheus/querybuilder/components/LabelFilterItem.tsx

```diff
@@ -0,0 +1,162 @@
+import debounce from 'debounce-promise';
+import React, { useState } from 'react';
+
+import { SelectableValue, toOption } from '@grafana/data';
+import { selectors } from '@grafana/e2e-selectors';
+import { AccessoryButton, InputGroup } from '@grafana/experimental';
+import { AsyncSelect, Select } from '@grafana/ui';
+
+import { PROMETHEUS_QUERY_BUILDER_MAX_RESULTS } from '../components/MetricSelect';
+import { QueryBuilderLabelFilter } from '../shared/types';
+
+export interface Props {
+  defaultOp: string;
+  item: QueryBuilderLabelFilter;
+  onChange: (value: QueryBuilderLabelFilter) => void;
+  onGetLabelNames: (forLabel: QueryBuilderLabelFilter) => Promise<SelectableValue[]>;
+  onGetLabelValues: (forLabel: QueryBuilderLabelFilter) => Promise<SelectableValue[]>;
+  onDelete: () => void;
+  invalidLabel?: boolean;
+  invalidValue?: boolean;
+  getLabelValuesAutofillSuggestions: (query: string, labelName?: string) => Promise<SelectableValue[]>;
+}
+
+export function LabelFilterItem({
+  item,
+  defaultOp,
+  onChange,
+  onDelete,
+  onGetLabelNames,
+  onGetLabelValues,
+  invalidLabel,
+  invalidValue,
+  getLabelValuesAutofillSuggestions,
+}: Props) {
+  const [state, setState] = useState<{
+    labelNames?: SelectableValue[];
+    labelValues?: SelectableValue[];
+    isLoadingLabelNames?: boolean;
+    isLoadingLabelValues?: boolean;
+  }>({});
+
+  const isMultiSelect = (operator = item.op) => {
+    return operators.find((op) => op.label === operator)?.isMultiValue;
+  };
+
+  const getSelectOptionsFromString = (item?: string): string[] => {
+    if (item) {
+      if (item.indexOf('|') > 0) {
+        return item.split('|');
+      }
+      return [item];
+    }
+    return [];
+  };
+
+  const labelValueSearch = debounce((query: string) => getLabelValuesAutofillSuggestions(query, item.label), 350);
+
+  return (
+    <div data-testid="prometheus-dimensions-filter-item">
+      <InputGroup>
+        {/* Label name select, loads all values at once */}
+        <Select
```

Comment:
I tried to figure out how to call the API for label names with regex match and failed. I'll need to look into this more in another PR/issue.